Arrowgene.Ddon.Shared/Entity/Structure/CDataGPDetail.cs

```diff
@@ -0,0 +1,57 @@
+using Arrowgene.Buffers;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Arrowgene.Ddon.Shared.Entity.Structure
+{
+    public class CDataGPDetail
+    {
+        public uint GP { get; set; }
+        public uint Max { get; set; }
+        public bool isFree { get; set; }
+
+        public uint Type {  get; set; }
+
+        public ulong Expire { get; set; }
+
+        public ulong Created { get; set; }
```

Comment:
These two ulongs are probably timestamps. Check CDataBazaarItemInfo for an example on how to use timestamps on packets